enhancements/scheduling/prevent-user-workloads-from-being-scheduled-on-control-plane-nodes.md

```diff
@@ -0,0 +1,526 @@
+- As an administrator, I want only certain workloads to be allowed to schedule pods on tainted nodes.
+- As an administrator, I want to restrict workloads from being scheduled on special node groups (even those using `.spec.nodeName` in their pods), so that those node groups can only run specialized workloads (e.g., GPU-enabled nodes) without obstruction.
+- As a cluster administrator, I want to have a clear and manageable way to update and maintain who or what will be allowed to schedule workloads on control plane nodes, so that I can efficiently manage permissions as the cluster evolves or as new teams/services are on-boarded, per namespace or cluster-wide.
+- As an OpenShift developer, I want to understand the impact of the new scheduling restrictions on existing and future workloads, so that I can design applications that comply with cluster policies and make informed decisions about resource requests and deployment strategies.
+- As an end user, I want to receive informative feedback when my workloads are rejected due to being repelled from control plane nodes or special node groups, so that I can make the necessary adjustments without needing extensive support from cluster administrators.
+- As a security professional, I want to ensure that this pod rejection mechanism is enforceable and auditable, so that I can verify compliance with internal and external regulations regarding resource access and control plane integrity.
+- As an administrator, I want the ability to temporarily override scheduling restrictions for emergency or maintenance tasks without compromising the overall security posture of the cluster, ensuring that critical operations can be performed when necessary.
+
+### Goals
+
+<!-- *Summarize the specific goals of the proposal. How will we know that
+this has succeeded?  A good goal describes something a user wants from
+their perspective, and does not include the implementation details
+from the proposal.* -->
+
+- Enhance Cluster Security and Stability: Prevent non-control plane workloads from being scheduled on control plane nodes to avoid resource competition and potential out-of-memory (OOM) issues that could affect critical cluster operations.
+
+- Flexible and Manageable Workload Scheduling: Enable cluster administrators to specify and manage exceptions, allowing certain workloads to be scheduled on control plane nodes when necessary for operational requirements.
+
+- Protect Specialized Resources: Restrict scheduling of regular user workloads on nodes with specialized resources (e.g., GPU-enabled nodes) to ensure these costly resources are reserved for appropriate workloads.
+
+- Emergency Override Capability: Allow administrators to temporarily override scheduling restrictions for emergency or maintenance tasks, ensuring critical operations can be performed when necessary without compromising the cluster's overall security posture.
```

Comment:
Maybe I can phrase this better, but I only mean admins can disable the very thing we are proposing here, in case they actually need to schedule stuff on nodes that would be protected. This is a goal.

the previous one same as https://github.com/openshift/enhancements/pull/1583#discussion_r1654446908

Reply:
See my other two answers again.